src/facade/dragonfly_connection.cc

```diff
@@ -97,9 +97,9 @@ struct Connection::Shutdown {
   }
 };
 
-Connection::PubMessage::PubMessage(string pattern, shared_ptr<char[]> buf, size_t channel_len,
+Connection::PubMessage::PubMessage(string_view pattern, shared_ptr<char[]> buf, size_t channel_len,
```

Comment:
This is actually less ideal than the previous (existing) code.
Think of the case in which you have a temporary / disposable string. You can call this constructor with `std::move()` (i.e. `PubMessage msg(std::move(s), ...)`) and it will move the string inside without additional allocations.
Otherwise, it's roughly the same runtime performance as with `string_view`

See some discussion about this here: https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference